src/ray/gcs/gcs_client/accessor.cc

```diff
@@ -470,6 +470,25 @@ Status NodeInfoAccessor::AsyncRegister(const rpc::GcsNodeInfo &node_info,
   return Status::OK();
 }
 
+Status NodeInfoAccessor::AsyncCheckSelfAlive(
+    const std::function<void(Status, bool)> &callback) {
+  rpc::CheckAliveRequest request;
+  auto node_addr = local_node_info_.node_manager_address() + ":" +
+                   std::to_string(local_node_info_.node_manager_port());
+  RAY_CHECK(callback != nullptr);
+  request.add_raylet_address(node_addr);
```

Comment:
Maybe it doesn't need to be done here, but we probably should check the node id instead of address?

Reply:
I agree, but the existing code path only accept the address. I think it still should work because no two raylets can start on the same addresses. (port conflict?)

Reply:
When you restart the raylet, node id is guaranteed to be different but ip+port can be reused, not sure if it will cause issues.

(Not this PR) I think we should change the existing code path to also use node id?

Reply:
> When you restart the raylet, node id is guaranteed to be different but ip+port can be reused, not sure if it will cause issues.

I don't think it'll cause issue for this PR. Then this node is alive and GCS will report it's alive.


> (Not this PR) I think we should change the existing code path to also use node id?

I think it depends on the actual use case. Node id and ip+port both have useful cases. When you know a node has ip+port and you want to check whether the node is alive, just use ip+port.

node id is more internal concept for ray which might be more useful.